code/modules/mob/living/simple_animal/hostile/mining/goliath.dm

```diff
@@ -164,12 +203,27 @@
 
 /obj/effect/temp_visual/goliath_tentacle/original/Initialize(mapload, new_spawner)
 	. = ..()
-	var/list/directions = cardinal.Copy()
-	for(var/i in 1 to 3)
-		var/spawndir = pick_n_take(directions)
-		var/turf/T = get_step(src, spawndir)
-		if(T)
-			new /obj/effect/temp_visual/goliath_tentacle(T, spawner)
+	if(spawner.maxHealth <= 150) // Normal subadults only spawn 2 tentacles
```

Comment:
this seems like a lot of duplicate code, effectively the same thing three times.